### PR TITLE
Fix 10843: Change Array.prototype.findIndex method definition according to ES6 spec

### LINKED ISSUE #10843: Correct findIndex definition
Definition for array findIndex is not correct.

Currently:
```ts
findIndex(predicate: (value: T) => boolean, thisArg?: any): number;
```
Should be:
```ts
findIndex(predicate: (value: T, index: number, obj: Array<T>) => boolean, thisArg?: any): number;
```

Standard: http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.findindex

### PR DESCRIPTION
**Pull request by Salasar**
Fixes #10843 according to standart http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.findindex

### CODE REVIEW

File: lib/lib.es2015.core.d.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ interface Array<T> {
       * @param thisArg If provided, it will be used as the this value for each invocation of
       * predicate. If it is not provided, undefined is used instead.
       */
-    findIndex(predicate: (value: T) => boolean, thisArg?: any): number;
```
Comment (mhegazy) [inline lib/lib.es2015.core.d.ts]: these files are generated from the build and should not be changed manually. so please remove these.

see https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes for more details.

File: lib/lib.es6.d.ts [hunk 2]
```diff
@@ -4143,7 +4143,7 @@ interface Array<T> {
       * @param thisArg If provided, it will be used as the this value for each invocation of
       * predicate. If it is not provided, undefined is used instead.
       */
-    findIndex(predicate: (value: T) => boolean, thisArg?: any): number;
+    findIndex(predicate: (value: T, index: number, obj: Array<T>) => boolean, thisArg?: any): number;
```
Comment (mhegazy) [inline lib/lib.es6.d.ts]: same for this file.

**Comment by Salasar** [comment]
I have added necessary changes, please, check it.

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**